src/timelines.js

```diff
@@ -845,21 +843,31 @@ export default Kapsule({
         .transition().duration(state.transDuration)
         .call(state.xGrid);
 
-      if (
-        state.dateMarker &&
-        state.dateMarker >= state.xScale.domain()[0] &&
-        state.dateMarker <= state.xScale.domain()[1]
-      ) {
-        state.dateMarkerLine
-          .style('display', 'block')
-          .transition().duration(state.transDuration)
-            .attr('x1', state.xScale(state.dateMarker) + state.leftMargin)
-            .attr('x2', state.xScale(state.dateMarker) + state.leftMargin)
-            .attr('y1', state.topMargin + 1)
-            .attr('y2', state.graphH + state.topMargin)
-      } else {
-        state.dateMarkerLine.style('display', 'none');
-      }
+      if (typeof state.dateMarker != "undefined" && 
+        state.dateMarker != null && 
+        state.dateMarker.length != null && 
+        state.dateMarker.length > 0
+        ) {  
+        for (var i = 0; i < state.dateMarker.length; i++) {
+
+          state.dateMarkerLine.push(state.svg.append('line').attr('class', 'x-axis-date-marker'));
```

Comment:
Unfortunately this won't be able to react to changes in props. Every time the graph is re-rendered a new line will be added to the DOM. Similarly they will not be removed if the prop changes value. The maintenance of these objects should be done with an enter/update/exit d3 pattern.